exporters/neutron.go

```diff
@@ -486,3 +496,59 @@ func ListSubnetsPerPool(exporter *BaseOpenStackExporter, ch chan<- prometheus.Me
 
 	return nil
 }
+
+// ListNetworkDHCPAgentAssociation enumerate the network-dhcpagent association
+func ListNetworkDHCPAgentAssociation(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metric) error {
+	const dhcpAgentType = "DHCP agent"
+	allPagesAgents, err := agents.List(exporter.Client, agents.ListOpts{AgentType: dhcpAgentType}).AllPages()
+	if err != nil {
+		return err
+	}
+	dhcpAgents, err := agents.ExtractAgents(allPagesAgents)
+	if err != nil {
+		return err
+	}
+	for _, agent := range dhcpAgents {
+		networks, err := agents.ListDHCPNetworks(exporter.Client, agent.ID).Extract()
+		if err != nil {
+			return err
+		}
+
+		adminStateUp := "down"
+		if agent.AdminStateUp {
+			adminStateUp = "up"
+		}
+
+		for _, network := range networks {
+			ch <- prometheus.MustNewConstMetric(exporter.Metrics["network_dhcpagent_association"].Metric,
+				prometheus.GaugeValue, float64(1), network.ID, agent.ID, agent.Host,
+				adminStateUp, strconv.FormatBool(agent.Alive))
+		}
+	}
+	return nil
+}
+
+// ListDRAgentBGPSpeakerAssociation enumerate the BGP Agent and Speaker association
+func ListDRAgentBGPSpeakerAssociation(exporter *BaseOpenStackExporter, ch chan<- prometheus.Metric) error {
+	const bgpAgentType = "BGP dynamic routing agent"
+	allPagesAgents, err := agents.List(exporter.Client, agents.ListOpts{AgentType: bgpAgentType}).AllPages()
+	if err != nil {
+		return err
+	}
+	dragents, err := agents.ExtractAgents(allPagesAgents)
+	if err != nil {
+		return err
+	}
+	for _, agent := range dragents {
+		pages, err := agents.ListBGPSpeakers(exporter.Client, agent.ID).AllPages()
+		if err != nil {
+			return err
+		}
+		allSpeakers, err := agents.ExtractBGPSpeakers(pages)
```

Comment:
never shadow an error, please check this error code

Reply:
I've fixed it. Please check it out.